### DISCUSSION TITLE
What is the best way to set the duration of an animation in SwiftUI?

### DISCUSSION THREAD

**Discussion by imthath-m** [airbnb/lottie-ios#2302] [Q&A] [answered]
Here's what we have done now with the `.animationSpeed(_:)` modifier.

```swift
struct PlaybackDurationDemo: View {
  let animation: LottieAnimation? = LottieAnimation.named("check")

  var body: some View {
    LottieView {
      animation
    }
    .animationSpeed(animation!.speedToPlay(forDuration: 10))
    .playing()
  }
}

extension LottieAnimation {
  var timeInterval: TimeInterval {
    (endFrame - startFrame) / framerate
  }

  func speedToPlay(forDuration duration: TimeInterval) -> Double {
    timeInterval / duration
  }
}
```

Would love to know if this approach is correct or if there is an easier way.

**Comment by calda** [comment]
Your approach makes sense to me. Alternatively you could create a view modifier in an extension like `.duration(10, animation: animation)` that under the hood performs the same math and applies the `animationSpeed` modifier.

**Reply by calda** [reply, marked as answer]
Another option is to use a `.configure { ... }` with something like this:

```
.configure { animationView in
  guard let animation = animationView.animation else { return }
  let expectedSpeed = animation.speedToPlay(forDuration: 10)

  if animationView.animationSpeed != expectedSpeed {
    animationView.animationSpeed = expectedSpeed
  }
}
```

This could be wrapped up in a custom view modifier like `.duration(10)` and seems appealing since then you don't have to pass the animation into the view modifier.

**Reply by imthath-m** [reply]
Thanks for the response and suggestions. Not needing to pass the `animation` is exactly what we want.

Does it make sense to add it a `.duration(_:)` modifier to library? I can open a PR if that's acceptable.

**Reply by calda** [reply]
Since the animation duration is typically defined via the animation json itself, I think we can continue without it for now.
